src/pt/yugenmangas/src/eu/kanade/tachiyomi/extension/pt/yugenmangas/YugenMangas.kt

```diff
@@ -1,142 +1,198 @@
 package eu.kanade.tachiyomi.extension.pt.yugenmangas
 
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.randomua.PREF_KEY_RANDOM_UA
+import eu.kanade.tachiyomi.lib.randomua.addRandomUAPreferenceToScreen
+import eu.kanade.tachiyomi.lib.randomua.getPrefCustomUA
+import eu.kanade.tachiyomi.lib.randomua.getPrefUAType
+import eu.kanade.tachiyomi.lib.randomua.setRandomUserAgent
 import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.POST
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
 import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.getPreferencesLazy
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
-import okhttp3.Headers
-import okhttp3.HttpUrl.Companion.toHttpUrl
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody
 import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 import okio.Buffer
+import org.jsoup.nodes.Element
 import rx.Observable
 import uy.kohesive.injekt.injectLazy
-import java.util.concurrent.TimeUnit
 
-class YugenMangas : HttpSource() {
+class YugenMangas : HttpSource(), ConfigurableSource {
 
     override val name = "Yugen Mangás"
 
-    override val baseUrl = "https://yugenmangasbr.voblog.xyz"
+    override val baseUrl = "https://yugenmangasbr.readmis.com"
 
     override val lang = "pt-BR"
 
     override val supportsLatest = true
 
+    private val preferences by getPreferencesLazy()
```

Comment:
```suggestion
    private val preferences by getPreferencesLazy {
        getString(PREF_KEY_RANDOM_UA, "off").let {
            if (it != "off" || getPrefCustomUA().isNullOrBlank()) {
                return@let
            }
            edit().putString(PREF_KEY_RANDOM_UA, "mobile").apply()
        }
    }
```
and remove the `init`